src/lib/process/index.ts

```diff
@@ -2,54 +2,70 @@ import { createNodeApi } from '../utils/polkadot.js'
 import { Constants } from './constants.js'
 import { createProcessTransaction, disableProcessTransaction, getVersion } from './api.js'
 import { utf8ToHex } from './hex.js'
-import { mapRestrictions } from './map.js'
-import type { ChainRestrictions } from '../types/restrictions.js'
-import { VersionError } from '../types/error.js'
+import { stepValidation } from '../types/restrictions.js'
+import { NoValidRestrictionsError, VersionError } from '../types/error.js'
 
 export const defaultOptions: Polkadot.Options = {
   API_HOST: 'localhost',
   API_PORT: 9944,
   USER_URI: '//Alice',
 }
 
+const validate = (program: Process.Program): Process.Program => {
+  return program.reduce((out: Process.Program, step: Process.ProgramStep)=> {
+    if (!stepValidation.parse(step)) return out
+    out.push(step) // .push in reduce is more effective than spread, can cause issuecs
+    return out
+  }, [])
+}
+
+
+export const loadProcesses = async (data: string, res: Process.Response = {}): Promise<Process.Response|string> => {
+  try {
+    const processes: Process.CLIParsed = JSON.parse(data)
+    // TODO more elegant promise.series way.
+    for (let i = 0; i < processes.length; i++) {
+      const { name, version, program } = processes[i]
+      res[name] = await createProcess(name, version, validate(program))
+    }
+
+    return res
+  } catch (err) {
+    // TODO handle error
+    console.log('loadProcesses: ', err)
+    return 'error'
+  }
+}
+
 export const createProcess = async (
   name: string,
   version: number,
-  rawRestrictions: string,
+  program: Process.Program,
   dryRun: boolean = false,
```

Comment:
dryrun should be configurable as an argument?

Reply:
and it's now